File: 857-positions-of-large-groups/positions-of-large-groups.go
func largeGroupPositions(s string) [][]int {
    result := [][]int{}

	start := 0
	l := 0 // length
	for i := range len(s) {
		if s[i] == s[start] {
			l++
		} else {
			if l >= 3 {
				result = append(result, []int{start, start + l - 1})
			}
            
			start = i
			l = 1
		}
	}

	if l >= 3 {
		result = append(result, []int{start, start + l - 1})
	}

	return result
}

File: 857-positions-of-large-groups/positions-of-large-groups.py
class Solution:
    def largeGroupPositions(self, s: str) -> List[List[int]]:
        ans = []
        i = 0  
        n = len(s)
        
        for j in range(n):
            if j == n - 1 or s[j] != s[j + 1]:
                if j - i + 1 >= 3:
                    ans.append([i, j])
                
                i = j + 1
        
        return ans